src/advec_cell.cpp

```diff
@@ -61,82 +49,119 @@ void advec_cell_kernel(
 		if (sweep_number == 1) {
 
 
-			_Pragma("kernel2d")
+			double *volume = field.volume.data;
+			const int volume_sizex = field.volume.sizeX;
+			double *vol_flux_x = field.vol_flux_x.data;
+			const int vol_flux_x_sizex = field.vol_flux_x.sizeX;
+			double *vol_flux_y = field.vol_flux_y.data;
+			const int vol_flux_y_sizex = field.vol_flux_y.sizeX;
+			double *pre_vol = field.work_array1.data;
+			const int pre_vol_sizex = field.work_array1.sizeX;
+			double *post_vol = field.work_array2.data;
+			const int post_vol_sizex = field.work_array2.sizeX;
+
+			#pragma omp target teams distribute parallel for simd collapse(2) omp_use_target(use_target)
 			for (int j = (y_min - 2 + 1); j < (y_max + 2 + 2); j++) {
 				for (int i = (x_min - 2 + 1); i < (x_max + 2 + 2); i++) {
-					pre_vol(i, j) = volume(i, j) + (vol_flux_x(i + 1, j + 0) - vol_flux_x(i, j) + vol_flux_y(i + 0, j + 1) - vol_flux_y(i, j));
-					post_vol(i, j) = pre_vol(i, j) - (vol_flux_x(i + 1, j + 0) - vol_flux_x(i, j));
+					pre_vol[i + j * pre_vol_sizex] = volume[i + j * volume_sizex] +
+					                                 (vol_flux_x[(i + 1) + (j + 0) * vol_flux_x_sizex] - vol_flux_x[i + j * vol_flux_x_sizex] +
+					                                  vol_flux_y[(i + 0) + (j + 1) * vol_flux_y_sizex] -
+					                                  vol_flux_y[i + j * vol_flux_y_sizex]);
+					post_vol[i + j * post_vol_sizex] = pre_vol[i + j * pre_vol_sizex] - (vol_flux_x[(i + 1) + (j + 0) * vol_flux_x_sizex] - vol_flux_x[i + j * vol_flux_x_sizex]);
 				}
 			}
 
 
 		} else {
 
 
-			_Pragma("kernel2d")
+			double *volume = field.volume.data;
+			const int volume_sizex = field.volume.sizeX;
+			double *vol_flux_x = field.vol_flux_x.data;
+			const int vol_flux_x_sizex = field.vol_flux_x.sizeX;
+			double *pre_vol = field.work_array1.data;
+			const int pre_vol_sizex = field.work_array1.sizeX;
+			double *post_vol = field.work_array2.data;
+			const int post_vol_sizex = field.work_array2.sizeX;
+
+			#pragma omp target teams distribute parallel for simd collapse(2) omp_use_target(use_target)
 			for (int j = (y_min - 2 + 1); j < (y_max + 2 + 2); j++) {
 				for (int i = (x_min - 2 + 1); i < (x_max + 2 + 2); i++) {
-					pre_vol(i, j) = volume(i, j) + vol_flux_x(i + 1, j + 0) - vol_flux_x(i, j);
-					post_vol(i, j) = volume(i, j);
+					pre_vol[i + j * pre_vol_sizex] = volume[i + j * volume_sizex] + vol_flux_x[(i + 1) + (j + 0) * vol_flux_x_sizex] - vol_flux_x[i + j * vol_flux_x_sizex];
+					post_vol[i + j * post_vol_sizex] = volume[i + j * volume_sizex];
 				}
 			}
 
 		}
 
 		// DO k=y_min,y_max
 		//   DO j=x_min,x_max+2
-		_Pragma("kernel2d")
+		double *vertexdx = field.vertexdx.data;
+		double *density1 = field.density1.data;
+		const int density1_sizex = field.density1.sizeX;
+		double *energy1 = field.energy1.data;
+		const int energy1_sizex = field.energy1.sizeX;
```

Comment:
Are all these `sizeX` not the same size and so we can use just the one variable for them?
I expect that will break the abstraction in `field`...

Reply:
Addressed in https://github.com/UoB-HPC/cloverleaf_openmp_target/pull/2/commits/75fb6f7f6172db301a14991e51971363d27b8805